state/allcollections.go

```diff
@@ -383,8 +383,13 @@ func allCollections() CollectionSchema {
 
 		// -----
 
-		providerIDsC:          {},
-		spacesC:               {},
+		providerIDsC: {},
+		spacesC: {
+			indexes: []mgo.Index{
+				{Key: []string{"model-uuid", "spaceid"}},
```

Comment:
why use "spaceid" for this rather than using "_id" for this? If you use "_id" then it is the primary key for the document, and so naturally comes with an index on it, and also implies that it is unique. It is also what we do for "machines" "units", etc. (they all end up as modeluuid:identifier)
It feels more consistent to me, but if you have a strong feeling to do it differently, I can be swayed.

Reply:
I based it all off machines, which has the same form of index. We get one automatically for _id as I understand.